### DISCUSSION TITLE
Battery timing for charge.

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [nathanmarlor/foxess_em#99] [General]
Trying to get my head around the logic of how the battery charging period works.

Last night for example it set the system to change from 00:30 until 23:59 - so basically all day which means pulling from the grid.

My battery was at 99% after its overnight charge so just trying to work out why it would keep charging?

![image](https://user-images.githubusercontent.com/96144685/204083776-d0a4fa96-d343-4866-8343-d31f430f4cfb.jpeg)

**Comment by nathanmarlor** [comment]
To force charge it sets both charge switches to on, the start time to now (00:30) and the end time to the end of the day (23:59). Once the required charge is achieved, or at the end of the off-peak period it resets this by turning both charge switches off and setting the times to 00:00. 

I'm unsure what the problem you've experienced is? Did it continue to force charge after the off-peak period? If so can you include some logs please?

**Reply by Captain0Kangaroo** [reply]
> To force charge it sets both charge switches to on, the start time to now (00:30) and the end time to the end of the day (23:59). Once the required charge is achieved, or at the end of the off-peak period it resets this by turning both charge switches off and setting the times to 00:00. 
> 
> I'm unsure what the problem you've experienced is? Did it continue to force charge after the off-peak period? If so can you include some logs please?

Yeah not sure to be honest what it did.. it seemed to be charging from the grid constantly and not using the battery as I would expect to see the battery at about 85% at 9am. 

Where would I get the logs from?

**Reply by nathanmarlor** [reply]
Makes sense. Logs are in settings -> system -> logs. Most likely an issue with the Fox Cloud at the time it tried to stop the force charge, although it looks like it set the min SoC back ok?

**Reply by Captain0Kangaroo** [reply]
What does the call to setback the times happen?

I rebooted HA at about 01:30 this morning and wondered if it had anything to do with that?

**Reply by nathanmarlor** [reply]
Yep that'll be why...! The charge active bool would've been set back to False after the reboot so it didn't stop the charge, but it did reset the SoC. Will look at if it's possible to persist the status (and others) past a reboot.

https://github.com/nathanmarlor/foxess_em/blob/main/custom_components/foxess_em/charge/charge_service.py#L137-L146

**Reply by Captain0Kangaroo** [reply]
I've just been keeping an eye on what's happening and seen a few errors tonight. From what I can see the call to the cloud failed and didn't update as the time period are what I set it to this morning.

![image](https://user-images.githubusercontent.com/96144685/204115605-2c071c9a-5070-4204-ba83-b90c498cec52.png)

![image](https://user-images.githubusercontent.com/96144685/204115608-185b2965-4a33-4e8b-89f3-a72af4702e90.png)

**Reply by nathanmarlor** [reply]
I've woken up to the same, looks like the Fox cloud went down around 12:30 this morning. Second time I've seen this happen in the last 3-4 months.

Wonder if a better approach would be to leave the force charge setup 00:30-4:30 and cancel if it isn't needed. This way if the cloud goes down you end up with a potentially unnecessary full battery vs. a costly empty battery.

**Reply by kme77** [reply]
Think I had a similar issue last night. I have my eco period set to 5-7am so I can use eco7 directly overnight but also top up my battery at the end of the cheap period. However when I woke up this morning I found find my battery still trickle charging and the house pulling from the grid. Checked the cloud and it hadn't reset the force charge to the default settings. 
With the seemingly more regular FoxESS 'cloud-outs' I think your suggestion of leaving the FC window static and toggling it off and on might be 'safer'. 👍 
Are you planning on implementing that in the next update?

**Reply by Captain0Kangaroo** [reply]
> Think I had a similar issue last night. I have my eco period set to 5-7am so I can use eco7 directly overnight but also top up my battery at the end of the cheap period. However when I woke up this morning I found find my battery still trickle charging and the house pulling from the grid. Checked the cloud and it hadn't reset the force charge to the default settings. 
> 
> With the seemingly more regular FoxESS 'cloud-outs' I think your suggestion of leaving the FC window static and toggling it off and on might be 'safer'. 👍 
> 
> Are you planning on implementing that in the next update?

Mine also did this yesterday. I think because the cloud was down it wasn't able to change the period so it was set to 00:00 to 00:00 and both toggles off.
![image](https://user-images.githubusercontent.com/96144685/205483685-7e429912-1960-49eb-9533-9ec27d6d92ba.jpeg)

**Reply by kme77** [reply]
Yeah it definitely seems the flakey nature of the Foxcloud is the weak point in this chain. The sooner we can move to a wholly Modbus controlled system the better but hopefully in the meantime Nathan can code a workaround. 😁😉

**Reply by spmaid** [reply]
> Think I had a similar issue last night. I have my eco period set to 5-7am so I can use eco7 directly overnight but also top up my battery at the end of the cheap period. However when I woke up this morning I found find my battery still trickle charging and the house pulling from the grid. Checked the cloud and it hadn't reset the force charge to the default settings. With the seemingly more regular FoxESS 'cloud-outs' I think your suggestion of leaving the FC window static and toggling it off and on might be 'safer'. 👍 Are you planning on implementing that in the next update?

Ah yes, I forgot this was another symptom I had, when I got to 19%SoC it started a self protection charge of approx 1.5kW, this is to protect it from too low a SoC.  This has also now stopped, although that disappeared after I had a different firmware update, that had to be rolled back. 

I'd advise contacting Fox support and seeing if they can update your firmware.

**Reply by spmaid** [reply]
> Yeah it definitely seems the flakey nature of the Foxcloud is the weak point in this chain. The sooner we can move to a wholly Modbus controlled system the better but hopefully in the meantime Nathan can code a workaround. 😁😉

I note they had an emergency maintenance message on the login page last week.  I wasn't able to log in at all for a few days.  Since they've completed it has been a lot more stable.

**Reply by kme77** [reply]
So that's two recommendations for a battery firmware update! 😉 I've seen mine do the same with the self-protection spikes so I'll definitely look at contacting FoxESS about updating the FW. 👍
